# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libGLESv2/global_state.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libGLESv2/global_state.h
--- a/gfx/angle/checkout/src/libGLESv2/global_state.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libGLESv2/global_state.h	Fri Nov 08 17:06:00 2019 +0000
@@ -21,11 +21,15 @@
 class Debug;
 class Thread;
 
+std::mutex &GetGlobalMutex();
 Thread *GetCurrentThread();
 Debug *GetDebug();
 void SetContextCurrent(Thread *thread, gl::Context *context);
 }  // namespace egl
 
+#define ANGLE_SCOPED_GLOBAL_LOCK() \
+    std::lock_guard<std::mutex> globalMutexLock(egl::GetGlobalMutex())
+
 namespace gl
 {
 extern Context *gSingleThreadedContext;
@@ -54,16 +58,4 @@
 
 }  // namespace gl
 
-#if ANGLE_FORCE_THREAD_SAFETY == ANGLE_ENABLED
-namespace angle
-{
-std::mutex &GetGlobalMutex();
-}  // namespace angle
-
-#    define ANGLE_SCOPED_GLOBAL_LOCK() \
-        std::lock_guard<std::mutex> globalMutexLock(angle::GetGlobalMutex())
-#else
-#    define ANGLE_SCOPED_GLOBAL_LOCK()
-#endif
-
 #endif  // LIBGLESV2_GLOBALSTATE_H_